### ISSUE TITLE
`--group-by-contrasts` outputs the same pairs

### ISSUE THREAD

**Issue by valosekj** [ivadomed/templates#3]
When trying PR https://github.com/ivadomed/templates/pull/1 and  `--group-by-contrasts` flag:

https://github.com/ivadomed/templates/blob/b1129df5b83a5edd1ce58d71509a1c641e791e0f/bids_to_msd_dataset_conversion.py#L101-L102

using:

```
python bids_to_msd_dataset_conversion.py --path-data test_dataset --path-out test_dataset_msd --split 0.6 0.2 0.2 --label-suffix _gmseg-manual_crop --include-contrasts T1w T2star --common-label-contrast T1w --group-by-contrasts
```

I get the following `dataset.json`:

```json
    "test": [
        {
            "image": "test_dataset/sub-0002/ses-01/anat/sub-0002_ses-01_T1w.nii.gz",
            "label": "test_dataset/derivatives/labels/sub-0002/anat/sub-0002_ses-01_T1w_gmseg-manual_crop.nii.gz"
        },
        {
            "image": "test_dataset/sub-0002/ses-01/anat/sub-0002_ses-01_T2star.nii.gz",
            "label": "test_dataset/derivatives/labels/sub-0002/anat/sub-0002_ses-01_T1w_gmseg-manual_crop.nii.gz"
        },
        {
            "image": "test_dataset/sub-0002/ses-01/anat/sub-0002_ses-01_T2star.nii.gz",
            "label": "test_dataset/derivatives/labels/sub-0002/anat/sub-0002_ses-01_T1w_gmseg-manual_crop.nii.gz"
        }
```

Notice that the second and the third pairs are the same.

**Comment by naga-karthik** [comment]
Ah! This seems to be a corner case that my template failed to account for! I noticed that the `--group-by-contrasts` flag only worked in the particular case of a cross-sectional dataset without the `ses` entity. 

I would have to add another `if` condition to account for multiple contrasts within a cross-section dataset that also has the `ses` entity. I will work on this fix today!

**Comment by naga-karthik** [comment]
Hey @valosekj, apologies for the delay on this! This feature has now been added in commit https://github.com/ivadomed/templates/pull/1/commits/53d7826fab805eab2bef71161a681741f5f3a02a. 

The output: 
<img width="1157" alt="Screen Shot 2023-03-13 at 12 48 07 PM" src="https://user-images.githubusercontent.com/53445351/224636017-5bda1906-259f-41a5-b80b-9cf7309f02d9.png">

**Comment by naga-karthik** [comment]
This has been fixed in commit 1d9b9dea2f8efbb02cd7a2849a77ba7fe097d4a2! 

Correct output: 

<img width="1181" alt="Screen Shot 2023-03-14 at 10 13 21 PM" src="https://user-images.githubusercontent.com/53445351/225076639-a8416730-8dc3-4a02-b1d6-6ead153b53c8.png">
